lib/_http_outgoing.js

```diff
@@ -673,6 +673,34 @@ OutgoingMessage.prototype.setHeader = function setHeader(name, value) {
   return this;
 };
 
+OutgoingMessage.prototype.setHeaders = function setHeaders(obj) {
+  if (this._header) {
+    throw new ERR_HTTP_HEADERS_SENT('set');
+  }
+
+  let k;
+  if (ArrayIsArray(obj)) {
+    if (obj.length % 2 !== 0) {
+      throw new ERR_INVALID_ARG_VALUE('headers', obj);
+    }
+
+    for (let n = 0; n < obj.length; n += 2) {
+      k = obj[n + 0];
+      if (k) this.setHeader(k, obj[n + 1]);
+    }
+  } else if (obj) {
```

Comment:
This should validate the type, otherwise the result of `Object.keys()` can be unexpected.

Reply:
the weird type could be `string` which would be converted to ["s", "t", "r", "i", "n", "g"]